starfish/imagestack/parser/_tiledata.py

```diff
@@ -11,11 +11,18 @@ class TileData:
     Base class for a parser to implement that provides the data for a single tile.
     """
     @property
-    def tile_shape(self) -> Tuple[int, int]:
+    def tile_shape(self) -> Mapping[Axes, int]:
         raise NotImplementedError()
 
     @property
     def numpy_array(self) -> np.ndarray:
+        """Return the image data representing the tile.  The tile must be row-major.
```

Comment:
What happens if it isn't? Can we detect these problems? Can we emit a warning or throw an error?

Reply:
Not really.  We can only tell when the data is misshapen.